packages/@aws-cdk/aws-dynamodb/lib/table.ts

```diff
@@ -390,56 +450,90 @@ abstract class TableBase extends Resource implements ITable {
    */
   public abstract readonly tableStreamArn?: string;
 
+  /**
+   * Optional KMS encryption key associated with this table.
+   */
+  public abstract readonly encryptionKey?: kms.IKey;
+
   protected readonly regionalArns = new Array<string>();
 
+  /**
+   * Returns an ARN that represents all objects within the table that match
+   * the key pattern specified. To represent all keys, specify ``"*"``.
+   *
+   * If you specify multiple components for keyPattern, they will be concatenated::
+   *
+   *   arnForObjects('home/', team, '/', user, '/*')
+   *
+   */
+  public arnForObjects(keyPattern: string): string {
```

Comment:
That looks awfully generic (prone to mis-use?). Maybe this should be `private` and used as the mechanism to build stream / index / ... ARNs, possibly for specific public APIs?